File: app/domain/resource-service.go
package domain

func AppendLastResourceToResourceMap(mapping map[string][]Resource, key string) (map[string][]Resource, Resource) {
	resource := LastResource(mapping, key)
	if resource == (Resource{}) || resource.IsLastPicture() {
		resource = NewResource()
		mapping[key] = append(mapping[key], resource)
	}
	return mapping, resource
}

func AppendResourceMap(mapping map[string][]Resource, key string, value Resource) map[string][]Resource {
	if len(mapping[key]) == 0 {
		return mapping
	}
	lastIndex := len(mapping[key]) - 1
	mapping[key][lastIndex] = value
	return mapping
}

func LastResource(mapping map[string][]Resource, key string) Resource {
	if len(mapping[key]) == 0 {
		return Resource{}
	}
	lastIndex := len(mapping[key]) - 1
	return mapping[key][lastIndex]
}

File: src/domain/resource_service.py
from domain.resource import Resource

def createResource(key: str, resources: dict) -> tuple:
    resource = __lastResource(key, resources)
    if resource == None:
        resource = Resource()
        resources[key] = [resource]
    elif resource.isLast():
        resource = Resource()
        resources[key] = [*resources[key], resource]
    return resources, resource
    
def updateLastResourceByKey(key: str, resources: dict, resource: Resource) -> dict:
    if key not in resources:
        return resources
    index = len(resources[key]) - 1
    resources[key][index] = resource
    return resources

def __lastResource(key: str, resources: dict) -> Resource:
    if key not in resources:
        return None
    
    index = len(resources[key]) - 1
    return resources[key][index]

